pkg/abap/build/bfw.go

```diff
@@ -555,10 +555,8 @@ func (logging *logStruct) print() {
 
 // ******** parsing ********
 func (v Value) String() string {
-	return fmt.Sprintf(
-		`{ "value_id": "%s", "value": "%s" }`,
-		v.ValueID,
-		v.Value)
+	returnByteArray, _ := json.Marshal(v)
```

Comment:
What is the reason for ignoring the error?

Reply:
https://pkg.go.dev/encoding/json#Marshal
"The nil pointer exception is not strictly necessary but mimics a similar, necessary exception in the behavior of UnmarshalJSON."

We basically just use it for a fixed strucutre with 2 strings so I rated the complexity involved as low enough to practically rule out that the error is ever != nil. In additon the real checking of the parameters is done in the backend so any issue will be reported back anyway.

Reply:
After a nights sleep I remembered the boy scouts mindset to leave everything in better shape than it was so I overcame my laziness and replaced the complete parsing logic by json.Marshal instead of just the bit which caused the trouble. Thus now also the error can be easily propagated.